tests/src/test/scala/akka/kafka/scaladsl/TransactionsSpec.scala

```diff
@@ -73,6 +73,63 @@ class TransactionsSpec extends SpecBase(kafkaPort = KafkaPorts.TransactionsSpec)
       }
     }
 
+    "complete with partitioned source" in {
+      assertAllStagesStopped {
+        val maxPartitions = 2
+        val sourceTopic = createTopic(1, maxPartitions, 1)
+        val sinkTopic = createTopicName(2)
+        val group = createGroupId(1)
+
+        givenInitializedTopic(sourceTopic)
+        givenInitializedTopic(sinkTopic)
+
+        Await.result(produce(sourceTopic, 1 to 100), remainingOrDefault)
+
+        val consumerSettings = consumerDefaults.withGroupId(group)
+
+        def runTransactional =
+          Transactional
+            .partitionedSource(consumerSettings, TopicSubscription(Set(sourceTopic), None))
+            .mapAsyncUnordered(maxPartitions) {
+              case (tp, streamCompletePromise, source) =>
+                source
+                  .filterNot(_.record.value() == InitialMsg)
+                  .map { msg =>
+                    ProducerMessage.single(new ProducerRecord[String, String](sinkTopic, msg.record.value),
+                      msg.partitionOffset)
+                  }
+                  .runWith(Transactional.sink(producerDefaults, group + "-" + tp, streamCompletePromise))
```

Comment:
Not super important, but it would be nice to find a way to make creating a unique transactional id a little smarter when using the partitioned source.  A convenience function that takes the group and tp?  Group and tp are also present in the offset info passed along to the sink & flow, but it's too late to use it then unless we lazily begin the transaction on the first message that arrives.  We would also need to enforce this behaviour only occurs when using the partitioned source.  WDYT?

Reply:
That looks pretty good.  I was thinking that a transactional id would still be provided, but it would be appended with GroupTopicPartition info.  One way to differentiate that messages are coming from a partitioned source instead of having `sinkFromPartitioned` and `flowFromPartitioned` would be to mix in a trait with `TransactionalMessage` (i.e. `FromPartitionedSource`) that you could match on in the `TransactionalProducerStageLogic`.

Reply:
Ah, yes of course.  That looks like a good solution.  Could use a trait on the PartitionOffset type too.  Either way.

Reply:
Either is fine with me.  @ennru WDYT?